src/base/intersection.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include <math.h>
+
+#include "vector.h"
+
+/// A utility struct that holds intersection information.
+namespace gm {
+struct Intersection {
+  Intersection() : surfacePoint(Vector3f(0.0f)), tHit(INFINITY) {}
+
+  Intersection(const Vector3f &surfacePoint, float tHit)
+      : surfacePoint(surfacePoint), tHit(tHit) {}
+
+  Vector3f surfacePoint;
```

Comment:
We will add more information to the intersection in the future, right?

Reply:
Yes. For now there for now I only added the necessary components to make our current code work. Theoretically there isn't much more we need though